File: php/spartacus_search.php
<?php

function spartacus_search(array $haystack, $needle): array {
    $results = [];
    foreach ($haystack as $x) {
        echo $x . PHP_EOL;
        $results[] = $x;
    }
    return $results;
}

print_r(spartacus_search([1, 2, 3, 4, 5], 3));
print_r(spartacus_search(['a', 'b', 'c'], 'z'));

File: scala/SpartacusSearch.scala
object SpartacusSearch {
  def spartacusSearch[T](haystack: Seq[T], needle: T): Seq[T] = {
    haystack.foreach(println)
    haystack
  }

  def main(args: Array[String]): Unit = {
    println(spartacusSearch(Seq(1, 2, 3, 4, 5), 3))
    println(spartacusSearch(Seq("a", "b", "c"), "z"))
  }
}
